# Abhishek Sharma

### CS 2nd Year | Section : "I" | Roll no.: 01 

### Enrollment no.: 12019009001127

## Advanced Artificial Intelligence and Machine Learning Lab 2

#### Date : 09.02.2021 | Faculty : BM and MB

In [1]:
from sklearn.datasets import load_boston

In [2]:
boston = load_boston()

In [3]:
import pandas as pd

In [4]:
data = pd.DataFrame(boston.data, columns = boston.feature_names)

In [5]:
data['MEDV'] = pd.DataFrame(boston.target) 

In [6]:
x = data['RM']

In [7]:
y = data['MEDV']

In [8]:
pd.DataFrame ([x,y]).transpose().head()

RM  MEDV
0  6.575  24.0
1  6.421  21.6
2  7.185  34.7
3  6.998  33.4
4  7.147  36.2

### Linear Regression Model :

In [9]:
from sklearn.linear_model import LinearRegression

In [10]:
model1 = LinearRegression()

In [11]:
from sklearn.model_selection import train_test_split

In [18]:
x = pd.DataFrame(x)
y = pd.DataFrame(y)

In [19]:
x_train1, x_test1, y_train1, y_test1 = train_test_split(x,y, test_size = 0.2)

In [20]:
print (x_train1.shape)

(404, 1)


In [21]:
print (x_test1.shape)

(102, 1)


In [22]:
linearRegression_train = LinearRegression()

In [23]:
linearRegression_train.fit(x_train1, y_train1)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [24]:
yTestPredict = linearRegression_train.predict(x_test1)

In [25]:
print (yTestPredict)

[[16.12141963]
 [28.59864126]
 [19.05463012]
 [21.17256759]
 [15.8644314 ]
 [22.23596716]
 [24.67292451]
 [24.07919308]
 [26.73769201]
 [24.22097969]
 [36.14877821]
 [20.58769783]
 [24.40707462]
 [22.78539027]
 [21.58020409]
 [15.19980667]
 [28.44799299]
 [22.55498703]
 [18.36342039]
 [22.82969859]
 [21.10167429]
 [18.87739685]
 [27.73905994]
 [19.59519156]
 [17.20254253]
 [24.31845799]
 [33.21556773]
 [21.50044913]
 [11.47790818]
 [30.32666556]
 [21.10167429]
 [28.86449115]
 [20.78265442]
 [15.90873972]
 [31.24827852]
 [20.91557936]
 [40.0390483 ]
 [20.28640128]
 [21.93467062]
 [17.44180743]
 [17.47725409]
 [14.8098935 ]
 [14.25160872]
 [29.02400109]
 [21.1459826 ]
 [25.47933585]
 [19.02804513]
 [18.94829016]
 [23.86651317]
 [22.81197526]
 [23.69814157]
 [35.66138674]
 [21.1459826 ]
 [30.52162215]
 [19.5863299 ]
 [25.15145432]
 [19.73697817]
 [20.10030636]
 [21.35866252]
 [14.73013853]
 [22.27141382]
 [30.07853899]
 [19.45340495]
 [35.00562367]
 [18.75333357]
 [14.69469188]
 [26.94151

In [27]:
import numpy as np

In [28]:
from sklearn.metrics import mean_squared_error

In [29]:
np.sqrt(mean_squared_error(y_test1, yTestPredict))

6.5323142340152325

In [30]:
linearRegression_train.score(x_test1, y_test1)

0.496653381741159

**Here we have imported the following libraries :**

- Calculating the mean squared error - ```from sklearn.metrics import mean_squared_error```


- For Training, Testing and Spliting the data - ```from sklearn.model_selection import train_test_split```

### Ridge regression Model :

**Importing the Lasso Model :** ```from sklearn.linear_model import Ridge```

In [32]:
from sklearn.linear_model import Ridge

In [33]:
ridge_reg = Ridge(alpha=1)

In [34]:
ridge_reg.fit(x_train1, y_train1)

Ridge(alpha=1, copy_X=True, fit_intercept=True, max_iter=None, normalize=False,
      random_state=None, solver='auto', tol=0.001)

In [35]:
yRidgePredict = ridge_reg.predict(x_test1)

In [36]:
np.sqrt(mean_squared_error(y_test1, yRidgePredict))

6.536180202560016

### Lasso Regression Model :

**Importing the Lasso Regression Model :** ```from sklearn.linear_model import Lasso```

In [37]:
from sklearn.linear_model import Lasso

In [38]:
lasso_reg = Lasso(alpha = 0.1)

In [39]:
lasso_reg.fit(x_train1, y_train1)

Lasso(alpha=0.1, copy_X=True, fit_intercept=True, max_iter=1000,
      normalize=False, positive=False, precompute=False, random_state=None,
      selection='cyclic', tol=0.0001, warm_start=False)

In [40]:
yLassoPredict = lasso_reg.predict(x_test1)

In [41]:
np.sqrt(mean_squared_error(y_test1, yLassoPredict))

6.550912093026748

### ElasticNet Model :

**Importing the ElasticNet Regression Model :** ```from sklearn.linear_model import ElasticNet```

In [42]:
from sklearn.linear_model import ElasticNet

In [44]:
elsticnet_reg = ElasticNet(alpha = 0.1, l1_ratio = 0.5)

In [46]:
elsticnet_reg.fit(x_train1, y_train1)

ElasticNet(alpha=0.1, copy_X=True, fit_intercept=True, l1_ratio=0.5,
           max_iter=1000, normalize=False, positive=False, precompute=False,
           random_state=None, selection='cyclic', tol=0.0001, warm_start=False)

In [47]:
yElasticNetPredict = ridge_reg.predict(x_test1)

In [48]:
np.sqrt(mean_squared_error(y_test1, yElasticNetPredict))

6.536180202560016

**Results of various regression models based on the mean squared error value :**

|Sl. No.|Name of the Model|mean squared error value|
|:---:|:---:|:---:|
|01|**Linear Regression Model**|**6.532314234015233**|
|02|Ridge regression Model (alpha = 1)|6.536180202560016|
|03|Lasso Regression Model (alpha = 0.1)|6.550912093026748|
|04|ElasticNet Model (alpha = 0.1, l1_ratio = 0.5)|6.536180202560016|

###  Conclusion :

From this implementing of the models and checking the mean squared error values we got **Linear regression Model** is having the lowest mean squared error and **for this dataset Linear Regression model is the best fitted model.**

### Thank You!